subsys/spm/Kconfig

```diff
@@ -8,7 +8,7 @@ menu "SPM"
 
 config SPM
 	bool "Use Secure Partition Manager"
-	default y if BOARD_NRF9160_PCA10090NS
+	default y if TRUSTED_EXECUTION_NONSECURE
```

Comment:
So, this is absolutely the opposite of what we need.
SPM is a module for the Secure Firmware.

Reply:
This does not enable the SPM library. It enables that the parent image uses the SPM image as a child image.

Which is what we need.

IS_SPM enables the SPM library.

Reply:
Then, we're good.

So, we seriously need to re-work the option names and help-documentation.

I strongly recommend `config SPM `to be the option to enable the library. This is the way we do it in all Zephyr subsystems. IS_SPM is not really good.

I strongly-recommend the current config SPM to be renamed to USES_SPM, REQUIRES_SPM, NEEDS_SPM, or anything else that reflects the real meaning of it. And, please, add a reasonably good help text.

I know I might sound a bit bad, but it would really help clarify things better for me. Upstream developers are used to CONFIG_"some-subsys" options meaning "We are building an application with "some-subsys"

Reply:
Will you do this, @SebastianBoe or you would like me to open a PR for that?

Reply:
We will work on the naming.

I can't guarantee it will happen before code freeze though.

The naming is inherited from CONFIG_BOOTLOADER_MCUBOOT. Before and after multi-image CONFIG_BOOTLOADER_MCUBOOT means "this image integrates with/uses an MCUBoot image.".

Introducing a new naming scheme would require changing the name of CONFIG_BOOTLOADER_MCUBOOT, which might make sense, but I need to look into what the consequences are.